endpoints/users.py

```diff
@@ -1,71 +1,210 @@
-        print(msg.key.decode('utf-8'), flush=True)
-        if msg.key.decode('utf-8') == name:
-            print("Got a response for the user {} --> {}".format(name, msg.value), flush=True)
-
-            return jsonify({'response': msg.value.decode('utf-8')})
-
-@app.route('/test/<string:name>')
-def test(name):
-    return f"It's working {name}!!"
-
-@app.route('/')
-def basic():
-    return "Hello world!"
-    
-
-if __name__ == "__main__":
-    app.run()
+from flask import Flask, Response
+
+# Import the messages to be sent to the statefun cluster
+from users_pb2 import CreateUserRequest
+from users_pb2 import UserRequest, UserResponse
+
+from kafka import KafkaProducer, KafkaConsumer
+from kafka.errors import NoBrokersAvailable
+
+import threading
+import time
+import uuid
+import json
+
+from typing import Dict
+
+# create the logger and configure
+import logging
+
+FORMAT = '[%(asctime)s] %(levelname)-8s %(message)s'
+logging.basicConfig(level=logging.INFO, format=FORMAT)
+
+logger = logging.getLogger()
+
+
+
+
+# Some parameters to send and read from kafka
+KAFKA_BROKER = "kafka-broker:9092"
+USER_TOPIC = "users"
+USER_CREATION_TOPIC = "users-create"
+USER_EVENTS_TOPIC = "user-events"
+
+brokers_available = False
+producer = None
+
+# Where yet to answer request messages 
+# are kept
+new_messages : Dict[str, str] = {}
+
+# The worker id that we'll use to identify 
+# messages adderessed to us
+WORKER_ID : str = str(uuid.uuid4())
+logger.info(f'Started worker with id {WORKER_ID}')
+
+
+# Get the global 
+while not brokers_available:
+    try:
+        # Try to create the producer and consumer for 
+        # this worker
+        producer = KafkaProducer(bootstrap_servers=[KAFKA_BROKER])
+        consumer = KafkaConsumer(
+            USER_EVENTS_TOPIC,
+            bootstrap_servers=[KAFKA_BROKER],
+            auto_offset_reset='latest',
+        )
+        brokers_available = True
+        logger.info('Got the broker!')
+    except NoBrokersAvailable:
+        time.sleep(2)
+        continue
+
+
+# Background thread receiving messages from
+# the kafka topic and putting them in the dictionary
+def consume_forever(cons: KafkaConsumer):
+    logger.info('Started consumer thread')
+    for msg in cons:
+        if msg.key.decode('utf-8') == WORKER_ID:
+            logger.info(f'Received message! {msg.value}')
+            resp = UserResponse()
+            resp.ParseFromString(msg.value)
+            new_messages[resp.request_id] = resp.result
+
+# Create and start the background thread for the consumer
+con_thread = threading.Thread(target=consume_forever, args=[consumer])
+con_thread.start()
+
+# Looks for the message in the local dict
+def get_message(id: str):
+    """ Returns a given message from the dictionary"""
+    logger.info(new_messages)
+
+    result = None
+    while id not in new_messages:
+        logger.info("Not found the message")
+        logger.info(new_messages)
+        time.sleep(0.01)

```

Comment:
Maybe we need to tune this a bit later, otherwise we might be burning our CPU's (throwback to Orleans)..

Reply:
Yes indeed ahahha. I didn't want to set anything cause in my computer runs much slower than in deployment but we should tune that later based on real-world delays